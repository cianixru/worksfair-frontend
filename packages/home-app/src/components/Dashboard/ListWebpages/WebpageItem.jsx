import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Image, Transformation } from 'cloudinary-react';

import { textColourHelper } from '../../../utils/helpers';

const WebpageItem = ({
  nextUpdate, dayCreated, username, webpage, sampleImage
}) => (
  <li>
    <div className="columns box">
      <div className="column is-3">
        <figure className="image is-180x180">
          <Image
            cloudName="worksfair"
            publicId={(webpage.featured_images && webpage.featured_images[0]) || sampleImage}
            type="fetch"
            className="offering-image">
            <Transformation width="180" fetchFormat="auto" />
          </Image>
        </figure>
      </div>
      <div className="column is-7">
        <div className="content">
          <h3>
            <a
              href={`http://${webpage.sub_domain_name}.worksfair.com`}
              className={textColourHelper(webpage.colour)}
              target="_blank"
              rel="noopener noreferrer"
            >
              {webpage.title}
            </a>
          </h3>
          { webpage.address && <span>
              {`${webpage.address}, ${webpage.city}, ${webpage.state}`}
            </span>
          }
        </div>
        <div>
          <h6 className="is-size-7">
            <strong>Day Created:</strong>
            <span className="has-text-black"> {dayCreated}</span>
          </h6>
          <h6 className="is-size-7">
            <strong>Due For Update:</strong>
            <span className="has-text-warning"> {nextUpdate}</span>
          </h6>
        </div>
      </div>
      <div className="column is-2 is-mobile">
        <Link
          to={`/dashboard/${username}/business/edit/${webpage.sub_domain_name}`}
          className="button is-white">
          <i className="fa fa-edit is-size-5 has-text-grey"/>
        </Link>
      </div>
    </div>
  </li>
);

WebpageItem.propTypes = {
  webpage: PropTypes.object,
  nextUpdate: PropTypes.string,
  dayCreated: PropTypes.string,
  sampleImage: PropTypes.string,
  username: PropTypes.string,
};

export default WebpageItem;
