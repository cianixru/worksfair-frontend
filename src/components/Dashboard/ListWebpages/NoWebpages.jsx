import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NoWebpages = ({ user }) => (
  <li>
    <div className="message is-warning">
      <div className="message-body">
        <p className="is-size-5">
          You have no active webpages
        </p>
        { user.confirmed_account &&
          <Link
            to={`/dashboard/${user.username}/webpages/new/basic-info`}
            className="button is-link is-outlined  margin-top-25">
            Create a Webpage
          </Link>
        }
      </div>
    </div>
  </li>
);

NoWebpages.propTypes = {
  user: PropTypes.object,
};

export default NoWebpages;
