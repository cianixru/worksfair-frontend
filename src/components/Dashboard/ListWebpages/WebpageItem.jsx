import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


import { textColourHelper } from '../../../utils/helpers';

const WebpageItem = ({
  nextUpdate, dayCreated, webpage, sampleImage
}) => (
  <li>
    <div className="media box">
      <div className="media-left">
        <figure className="image is-180x180">
          <img
            src={(webpage.featured_images && webpage.featured_images[0])
              || sampleImage}
            alt="sample" />
        </figure>
      </div>
      <div className="media-content">
        <div className="content">
          <h3>
            <Link
              to={`/webpage/${webpage.sub_domain_name}`}
              className={textColourHelper(webpage.colour)}
            >
              {webpage.title}
            </Link>
          </h3>
          <span>
            {`${webpage.address}, ${webpage.city}, ${webpage.state}`}
          </span>
        </div>
        <div>
          <h6 className="is-size-7">
            <strong>Day Created:</strong>
            <span className="has-text-black"> {dayCreated}</span>
          </h6>
          <h6 className="is-size-7">
            <strong>Due For Update:</strong>
            <span className="has-text-warning"> {nextUpdate}</span>
          </h6>
        </div>
      </div>
      <div>
        <Link to="/" className="button is-white">
          <i className="fa fa-edit is-size-5 has-text-grey"/>
        </Link>
      </div>
    </div>
  </li>
);

WebpageItem.propTypes = {
  webpage: PropTypes.object,
  nextUpdate: PropTypes.string,
  dayCreated: PropTypes.string,
  sampleImage: PropTypes.string,
};

export default WebpageItem;
