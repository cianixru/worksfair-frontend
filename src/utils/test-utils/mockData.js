// eslint-disable-next-line import/prefer-default-export
export const user = {
  username: 'Theookafor2',
  first_name: 'eod',
  last_name: 'okafor',
  email: 'theodorehope@andela.com',
  facebook: null,
  instagram: null,
  twitter: null,
  headline: null,
  current_location: null,
  role: 'admin',
  confirmed_account: false,
  verified_account: false,
  reset_token: null,
  image_url: null,
  created_at: '2019-03-10T08:48:42.942532Z',
  modified_at: '2019-03-10T08:48:42.942544Z',
  token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9',
};

export const webpage = {
  owner: {
    id: 1,
    username: 'theowa',
    first_name: 'theo',
    last_name: 'IO',
    email: 'theo.iowa@yahoo.com',
    facebook: null,
    instagram: null,
    twitter: null,
    headline: null,
    current_location: null,
    role: 'admin',
    confirmed_account: false,
    verified_account: false,
    reset_token: null,
    image_url: 'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    created_at: '2019-04-25T09:07:08.209916Z',
    modified_at: '2019-04-25T09:07:08.209930Z'
  },
  title: 'Happiness',
  sub_domain_name: 'happiness',
  description: 'Happiness truly lives here. We just happy',
  keywords: 'passion, software, engineering, technology, worksfair',
  country: 'Nigeria',
  state: 'Lagos',
  city: 'Okota',
  area: 'Omagba Phase 1, Phase 2, Abatete',
  address: 'No 7 Obinna Uzor Street, Canal Estate',
  phone: null,
  email: null,
  website: null,
  featured_images: [
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png'
  ],
  is_active: true,
  logo: null,
  colour: 'is-dark',
  created_at: '2019-04-28T18:39:25.174272Z',
  updated_at: '2019-05-01T12:18:49.669633Z'
};
