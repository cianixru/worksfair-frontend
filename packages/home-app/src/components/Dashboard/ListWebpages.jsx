import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';

import { getDateForNextUpdate } from '../../utils/helpers';
import WebpageItem from './ListWebpages/WebpageItem';
import NoWebpages from './ListWebpages/NoWebpages';
import {
  getCurrentUser,
  confirmAccount,
  CONFIRM_USER,
} from '../../actions/auth';
import alert from '../utils/alert';

const sampleImage = "https://imgplaceholder.com/180x180/131111?text=ADD+A+PICTURE&font-size=18";
class ListWebpages extends Component {
  async componentDidMount() {
    const { actions, location, user } = this.props;
    try {
      const params = new URLSearchParams(location.search);
      const token = params.get('token');
      if (token){
        const response = await actions.confirmAccount({
          token,
        });
        if (response.type === CONFIRM_USER) {
          if (response.data.message === 'Your account has already been activated.') {
            alert.info(response.data.message);
          } else {
            alert.success(response.data.message);
          }
        } else {
          alert.error(response.data.message);
        }
        setTimeout(() => (
          window.location.replace(`/dashboard/${user.username}/webpages`)
        ), 3000);
      }
      await actions.getCurrentUser();
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    const { user } = this.props;
    const username = user && user.username;
    return (
      <div>
        <div className="tabs is-boxed">
          <ul>
            <li className="is-active">
              <Link to={`/dashboard/${user && user.username}/webpages`}>
                Active Webpages ({
                  (user && user.webpages) && user.webpages.length})
              </Link>
            </li>
          </ul>
        </div>
        <div>
          <ul>
            { (user && user.webpages) && user.webpages.length > 0
              ? user.webpages.map((webpage) => {
                const dayCreated = new Date(webpage.created_at).toDateString();
                const nextUpdate = getDateForNextUpdate(webpage.updated_at);

                const itemProps = {
                  nextUpdate, dayCreated, webpage, sampleImage, username
                };
                return (
                  <WebpageItem key={webpage.sub_domain_name} {...itemProps} />
                );
              })
              : <NoWebpages user={user && user} />
            }
          </ul>
        </div>
      </div>
    );
  }
}

ListWebpages.propTypes = {
  user: PropTypes.object,
};

const mapStateToProps = ({ auth: { currentUser } }) => ({
  user: currentUser.user,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      getCurrentUser,
      confirmAccount,
    },
    dispatch,
  ),
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(ListWebpages));
