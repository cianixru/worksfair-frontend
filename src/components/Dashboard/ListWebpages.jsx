import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import sampleImage from '../../assets/sample-02.jpg';
import { getDateForNextUpdate } from '../../utils/helpers';
import WebpageItem from './ListWebpages/WebpageItem';
import NoWebpages from './ListWebpages/NoWebpages';

class ListWebpages extends Component {
  render() {
    const { user } = this.props;
    return (
      <div>
        <div className="tabs is-boxed">
          <ul>
            <li className="is-active">
              <Link to={`/dashboard/${user && user.username}/webpages`}>
                Active Webpages ({
                  (user && user.webpages) && user.webpages.length})
              </Link>
            </li>
            <li>
              <Link to="/">
                Inactive Webpages
              </Link>
            </li>
          </ul>
        </div>
        <div>
          <ul>
            { (user && user.webpages) && user.webpages.length > 0
              ? user.webpages.map((webpage) => {
                const dayCreated = new Date(webpage.created_at).toDateString();
                const nextUpdate = getDateForNextUpdate(webpage.updated_at);

                const itemProps = {
                  nextUpdate, dayCreated, webpage, sampleImage,
                };
                return (
                  <WebpageItem key={webpage.sub_domain_name} {...itemProps} />
                );
              })
              : <NoWebpages username={user && user.username} />
            }
          </ul>
        </div>
      </div>
    );
  }
}

ListWebpages.propTypes = {
  user: PropTypes.object,
};

const mapStateToProps = ({ auth: { currentUser } }) => ({
  user: currentUser.user,
});

export default withRouter(connect(
  mapStateToProps,
  null,
)(ListWebpages));
