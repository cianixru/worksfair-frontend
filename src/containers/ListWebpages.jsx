import React, { Component, Fragment } from 'react';
import { Helmet } from 'react-helmet';

import ListWebpages from '../components/Dashboard/ListWebpages';
import DashboardHeader from '../components/Dashboard/DashboardHeader';
import Footer from '../components/Footer/Footer';

class ListWebpagesContainer extends Component {
  render() {
    const username = localStorage.getItem('username');
    const navigation = [
      {
        text: 'Dashboard',
        to: `dashboard/${username}/webpages`,
      },
    ];
    return (
      <Fragment>
        <Helmet>
          <title>Dashboard - Worksfair</title>
        </Helmet>
        <div className="dashboard-content">
          <DashboardHeader
            title={`How far, ${username}!`}
            navigation={navigation}
          />
          <ListWebpages />
        </div>
        <Footer />
      </Fragment>
    );
  }
}

export default ListWebpagesContainer;
