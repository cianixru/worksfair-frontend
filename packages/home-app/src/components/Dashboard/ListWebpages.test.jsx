import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import ListWebpages from './ListWebpages';
import { user, webpage } from '../../utils/test-utils/mockData';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: { user, } },
  webpage: { newWebpage: webpage }
});

describe('ListWebpages.jsx', () => {
  test('Should display the ListWebpages Component', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <ListWebpages user={user} />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
