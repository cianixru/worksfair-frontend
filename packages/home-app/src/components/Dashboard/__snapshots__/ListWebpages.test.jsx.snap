// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListWebpages.jsx Should display the ListWebpages Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="tabs is-boxed"
        >
          <ul>
            <li
              class="is-active"
            >
              <a
                href="/dashboard/TheoOkafor/businesses"
              >
                Active Businesses (
                1
                )
              </a>
            </li>
          </ul>
        </div>
        <div>
          <ul>
            <li>
              <div
                class="columns box"
              >
                <div
                  class="column is-3"
                >
                  <figure
                    class="image is-180x180"
                  >
                    <img
                      class="offering-image"
                      src="http://res.cloudinary.com/worksfair/image/fetch/f_auto/https://res.cloudinary.com/worksfair/image/upload/v1558455846/dev/achebe_jato7y.jpg"
                    />
                  </figure>
                </div>
                <div
                  class="column is-7"
                >
                  <div
                    class="content"
                  >
                    <h3>
                      <a
                        class="has-text-info"
                        href="http://ideosynergy-ltd.worksfair.com"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        Ideosynergy LTD.
                      </a>
                    </h3>
                    <span>
                      7, Ziks avenue, Onitsha, Anambra
                    </span>
                  </div>
                  <div>
                    <h6
                      class="is-size-7"
                    >
                      <strong>
                        Day Created:
                      </strong>
                      <span
                        class="has-text-black"
                      >
                         
                        Tue May 21 2019
                      </span>
                    </h6>
                    <h6
                      class="is-size-7"
                    >
                      <strong>
                        Due For Update:
                      </strong>
                      <span
                        class="has-text-warning"
                      >
                         
                        Invalid Date
                      </span>
                    </h6>
                  </div>
                </div>
                <div
                  class="column is-2 is-mobile"
                >
                  <a
                    class="button is-white"
                    href="/dashboard/TheoOkafor/business/edit/ideosynergy-ltd"
                  >
                    <i
                      class="fa fa-edit is-size-5 has-text-grey"
                    />
                  </a>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="tabs is-boxed"
      >
        <ul>
          <li
            class="is-active"
          >
            <a
              href="/dashboard/TheoOkafor/businesses"
            >
              Active Businesses (
              1
              )
            </a>
          </li>
        </ul>
      </div>
      <div>
        <ul>
          <li>
            <div
              class="columns box"
            >
              <div
                class="column is-3"
              >
                <figure
                  class="image is-180x180"
                >
                  <img
                    class="offering-image"
                    src="http://res.cloudinary.com/worksfair/image/fetch/f_auto/https://res.cloudinary.com/worksfair/image/upload/v1558455846/dev/achebe_jato7y.jpg"
                  />
                </figure>
              </div>
              <div
                class="column is-7"
              >
                <div
                  class="content"
                >
                  <h3>
                    <a
                      class="has-text-info"
                      href="http://ideosynergy-ltd.worksfair.com"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Ideosynergy LTD.
                    </a>
                  </h3>
                  <span>
                    7, Ziks avenue, Onitsha, Anambra
                  </span>
                </div>
                <div>
                  <h6
                    class="is-size-7"
                  >
                    <strong>
                      Day Created:
                    </strong>
                    <span
                      class="has-text-black"
                    >
                       
                      Tue May 21 2019
                    </span>
                  </h6>
                  <h6
                    class="is-size-7"
                  >
                    <strong>
                      Due For Update:
                    </strong>
                    <span
                      class="has-text-warning"
                    >
                       
                      Invalid Date
                    </span>
                  </h6>
                </div>
              </div>
              <div
                class="column is-2 is-mobile"
              >
                <a
                  class="button is-white"
                  href="/dashboard/TheoOkafor/business/edit/ideosynergy-ltd"
                >
                  <i
                    class="fa fa-edit is-size-5 has-text-grey"
                  />
                </a>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
